tests-fuzz/targets/migration/fuzz_migrate_mito_regions.rs

```diff
@@ -0,0 +1,305 @@
+            .execute(sql.as_str())
+            .await
+            .context(error::ExecuteQuerySnafu { sql: &sql })?;
+        ensure!(
+            result.rows_affected() == input.rows as u64,
+            error::AssertSnafu {
+                reason: format!(
+                    "expected rows affected: {}, actual: {}",
+                    input.rows,
+                    result.rows_affected(),
+                )
+            }
+        );
+        if rng.gen_bool(0.2) {
+            flush_memtable(&ctx.greptime, &create_expr.table_name).await?;
+        }
+        if rng.gen_bool(0.1) {
+            compact_table(&ctx.greptime, &create_expr.table_name).await?;
+        }
+    }
+
+    // Fetches region distribution
+    let partitions = fetch_partitions(&ctx.greptime, table_ctx.name.clone()).await?;
+    let num_partitions = partitions.len();
+    let region_distribution = region_distribution(partitions);
+    info!("Region distribution: {region_distribution:?}");
+    let datanodes = fetch_nodes(&ctx.greptime)
+        .await?
+        .into_iter()
+        .flat_map(|node| {
+            if node.peer_type == PEER_TYPE_DATANODE {
+                Some(node)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+    info!("List datanodes: {:?}", datanodes);
+
+    // Generates region migration task.
+    let mut migrations = Vec::with_capacity(num_partitions);
+    let mut new_distribution: BTreeMap<u64, HashSet<_>> = BTreeMap::new();
+    for (datanode_id, regions) in region_distribution {
+        let step = rng.gen_range(1..datanodes.len());
+        for region in regions {
+            let to_peer = (datanode_id + step as u64) % datanodes.len() as u64;
+            new_distribution.entry(to_peer).or_default().insert(region);
+            migrations.push(Migration {
+                from_peer: datanode_id,
+                to_peer,
+                region_id: region,
+            })
+        }
+    }
+
+    let mut procedure_ids = Vec::with_capacity(migrations.len());
+    // Triggers region migrations
+    for Migration {
+        from_peer,
+        to_peer,
+        region_id,
+    } in &migrations
+    {
+        let procedure_id =
+            migrate_region(&ctx.greptime, region_id.as_u64(), *from_peer, *to_peer, 120).await;
+        info!("Migrating region: {region_id} from {from_peer} to {to_peer}, procedure: {procedure_id}");
+        procedure_ids.push(procedure_id);
+    }
+    info!("Excepted new region distribution: {new_distribution:?}");
+
+    // Waits for all region migrated
+    wait_for_region_distribution(
+        &ctx.greptime,
+        Duration::from_secs(120),
+        table_ctx.name.clone(),
+        new_distribution,
+    )
+    .await;
+
+    for procedure_id in procedure_ids {
+        wait_for_procedure_finish(&ctx.greptime, Duration::from_secs(120), procedure_id).await;
+    }
```

Comment:
Should we wait procedure before waiting region distribution?

Reply:
it's okay